File: c++/advanced/1_Kadane_algo.cpp
//Find largest sum subarray (non-empty)

int kadanes(vector<int>& nums) {
    int max_sum = INT_MIN;
    int curr_sum = 0;

    for (int n : nums) {
        curr_sum = max(n, curr_sum + n);
        max_sum = max(max_sum, curSum);
    }
    return max_sum;
}

File: js/advanced/1_Kadane_algo.js
//Find largest sum subarray (non-empty)

function kadanes(nums) {
    let maxSum = Number.MIN_SAFE_INTEGER;
    let curSum = 0;

    for (let n of nums) {
        curSum = Math.max(curSum+ n, n);
        maxSum = Math.max(maxSum, curSum);
    }
    return maxSum;
}